apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/internal/ChannelWrapper.kt

```diff
@@ -0,0 +1,33 @@
+package com.apollographql.apollo3.internal
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.Channel
+
+/**
+ * A wrapper for [Channel] that adds a method equivalent to [invokeOnClose], which is marked [ExperimentalCoroutinesApi].
```

Comment:
Is this copied from `kotlinx.coroutines` ? Maybe add a link to the source if it's the case?

Reply:
This is not exactly copied (I just took `invokeOnClose` + simplified it) but good point I'll link to it 👍